lib/tag-field.jsx

```diff
@@ -18,12 +23,30 @@ export default React.createClass( {
 
 	getInitialState: function() {
 		return {
-			selectedTag: -1
+			selectedTag: -1,
+			tagInput: '',
 		};
 	},
 
-	componentWillReceiveProps: function() {
-		this.setState( { selectedTag: -1 } );
+	componentWillReceiveProps: function( nextProps ) {
+		const { selectedTag } = this.state;
+		const { tags: prevTags } = this.props;
+		const { tags: nextTags } = nextProps;
+
+		// if the tags changed externally, we need
+		// to match up the selected index with where
+		// the tag has moved in the list
+
+		if (
+			-1 === selectedTag || // no tag is selected
+			prevTags === nextTags // the tags are identical
+		) {
+			return;
+		}
+
+		this.setState( {
+			selectedTag: nextTags.indexOf( prevTags[ selectedTag ] ),
+		} );
```

Comment:
this previously reset the tag selection whenever the props changed. I think this was to prevent switching the selection unexpectedly (index vs name)

for example, tag `Simplenote` was selected at index `2`, a new tag was added externally and that new tag comes before `Simplenote` in the list. Selected tag is still index `2` but is now a different tag, maybe `Electron`.

we don't need to be this cautious. If the tags don't change at all or if we don't even have a selected tag we can skip this. if we do have a selected tag, we just need to find where it ends up in the new list and update the selection index.